src/libraries/System.Security.Cryptography.Encoding/tests/Cbor.Tests/CborReaderTests.Array.cs

```diff
@@ -0,0 +1,272 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+using System;
+using Test.Cryptography;
+using Xunit;
+
+namespace System.Security.Cryptography.Encoding.Tests.Cbor
+{
+    public partial class CborReaderTests
+    {
+        // Data points taken from https://tools.ietf.org/html/rfc7049#appendix-A
+        // Additional pairs generated using http://cbor.me/
+
+        [Theory]
+        [InlineData(new object[] { }, "80")]
+        [InlineData(new object[] { 42 }, "81182a")]
+        [InlineData(new object[] { 1, 2, 3 }, "83010203")]
+        [InlineData(new object[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25 }, "98190102030405060708090a0b0c0d0e0f101112131415161718181819")]
+        [InlineData(new object[] { 1, -1, "", new byte[] { 7 } }, "840120604107")]
+        [InlineData(new object[] { "lorem", "ipsum", "dolor" }, "83656c6f72656d65697073756d65646f6c6f72")]
+        public static void ReadArray_SimpleValues_HappyPath(object[] expectedValues, string hexEncoding)
```

Comment:
It probably makes sense to put the hex encoding first; so that the cases can be better distinguished in the logs (xml or console).

Reply:
The order is intentional, for a couple of reasons. The data points are shared between reader and writer tests so this makes it easier to copy. I picked this ordering because it's just the natural `(input, encoding)` ordering.

Reply:
I used the same approach for integer and string tests.

Reply:
I guess my point is "make this test fail, can you tell which theory case it was?"

For integers and strings, the int/string value is easiest to match on.  For arrays the printed values may not make it as easy to match (or maybe it's fine).